mathics/builtin/specialfns/hypergeom.py

```diff
@@ -0,0 +1,217 @@
+    :sympy: https://docs.sympy.org/latest/modules/functions/special.html#sympy.functions.special.hyper.hyper</url>, <url>
+    :WMA: https://reference.wolfram.com/language/ref/HypergeometricPFQ.html</url>)
+    <dl>
+      <dt>'HypergeometricPFQ'[${a_1, ..., a_p}, {b_1, ..., b_q}, z$]
+      <dd>returns ${}_p F_q({a_1, ..., a_p}; {b_1, ..., b_q}; z)$.
+    </dl>
+    >> HypergeometricPFQ[{2}, {2}, 1]
+     = E
+
+    Result is symbollicaly simplified by default:
+    >> HypergeometricPFQ[{3}, {2}, 1]
+     = HypergeometricPFQ[{3}, {2}, 1]
+    unless a numerical evaluation is explicitly requested:
+    >> HypergeometricPFQ[{3}, {2}, 1] // N
+     = 4.07742
+
+    The following special cases are handled:
+    >> HypergeometricPFQ[{}, {}, z]
+     = 1
+    >> HypergeometricPFQ[{0}, {b}, z]
+     = 1
+    >> HypergeometricPFQ[{b}, {b}, z]
+     = E ^ z
+    """
+
+    attributes = A_NUMERIC_FUNCTION | A_PROTECTED | A_READ_PROTECTED
+    mpmath_name = "hyper"
+    nargs = {3}
+    rules = {
+        "HypergeometricPFQ[{}, {}, z_]": "1",
+        "HypergeometricPFQ[{0}, b_, z_]": "1",
+        "HypergeometricPFQ[b_, b_, z_]": "Exp[z]",
+    }
+    summary_text = "compute the generalized hypergeometric function"
+    sympy_name = "hyper"
+
+    def eval(self, a, b, z, evaluation: Evaluation):
+        "HypergeometricPFQ[a_, b_, z_]"
+        try:
+            a_sympy = [e.to_sympy() for e in a]
+            b_sympy = [e.to_sympy() for e in b]
+            return from_sympy(sympy.hyper(a_sympy, b_sympy, z.to_sympy()))
+        except Exception:
+            pass
+
+    def eval_N(self, a, b, z, evaluation: Evaluation):
+        "N[HypergeometricPFQ[a_, b_, z_]]"
+        try:
+            return run_mpmath(
+                mpmath.hyper,
+                tuple([a.to_python(), b.to_python(), z.to_python()]),
+                FP_MANTISA_BINARY_DIGITS,
+            )
+        except Exception:
+            pass
+
+
+class Hypergeometric1F1(MPMathFunction):
+    """
+    <url>
+    :Kummer confluent hypergeometric function: https://en.wikipedia.org/wiki/Confluent_hypergeometric_function</url> (<url>
+    :mpmath: https://mpmath.org/doc/current/functions/hypergeometric.html#hyper</url>, <url>
+    :WMA: https://reference.wolfram.com/language/ref/Hypergeometric1F1.html</url>)
+    <dl>
+      <dt>'Hypergeometric1F1'[$a$, $b$, $z$]
+      <dd>returns ${}_1 F_1(a; b; z)$.
+    </dl>
+
+    Result is symbollicaly simplified by default:
+    >> Hypergeometric1F1[3, 2, 1]
+     = HypergeometricPFQ[{3}, {2}, 1]
+    unless a numerical evaluation is explicitly requested:
+    >> Hypergeometric1F1[3, 2, 1] // N
+     = 4.07742
+
+    Plot 'M'[3, 2, x] from 0 to 2 in steps of 0.5:
+    >> Plot[Hypergeometric1F1[3, 2, x], {x, 0.5, 2}]
+     = -Graphics-
+    Here, plot explicitly requests a numerical evaluation.
+    """
+
+    attributes = A_LISTABLE | A_NUMERIC_FUNCTION | A_PROTECTED | A_READ_PROTECTED
```

Comment:
```suggestion
    attributes = A_LISTABLE | A_NUMERIC_FUNCTION | A_PROTECTED
```
In contrast to HypergeometricPFQ, WMA does not list this as being ReadProtected.

Reply:
@rocky I've these changes on my radar. Kindly wait until I've marked this PR for review. Before that, I'm just focusing on the important functional aspects. But, I'll fix the documentation and the attributes before completion.

Reply:
For now, I'll limit myself to these four functions, which I will also be needing at work right now. But, eventually, other hypergeometric functions can be implemented just via rules.

Reply:
I'll do that later in another PR.